dynamiq/nodes/tools/basetool.py

```diff
@@ -0,0 +1,45 @@
+from abc import ABC, abstractmethod
+from typing import Any, Literal
+
+from pydantic import BaseModel, Field, ValidationInfo, field_validator
+
+from dynamiq.nodes import NodeGroup
+from dynamiq.runnables import RunnableConfig
+
+
+class DefaultInputSchema(BaseModel, ABC):
+    input: str
+
+
+class ToolMixin:
```

Comment:
Should this be an abstract class?

Reply:
What about naming? `BaseTool`?

Reply:
[Here](https://github.com/dynamiq-ai/dynamiq/blob/c8f66ac6b8c27e3f5d91912fb8bf6051bd7f13e7/dynamiq/nodes/agents/base.py#L61), it should be: `tools: list[BaseTool] = []`